otherlibs/systhreads/st_stubs.c

```diff
@@ -376,6 +376,9 @@ CAMLprim value caml_thread_yield(value unit);
 
 void caml_thread_interrupt_hook(void)
 {
+  /* Do not attempt to yield from the backup thread */
+  if (caml_bt_is_in_blocking_section()) return;
```

Comment:
CI failure was hard tor reproduce, but when I managed to do it, the backtrace pointed fingers at this test not being enough to prevent `caml_thread_yield` from running from the backup thread. Do you have ideas (@Engil? @ctk21?) for a better way of excluding the backup thread here?
(To reproduce the bug reliably, it turns out that you either need an Intel CPU, or to put load on your AMD CPU and be patient. I'd be curious at an explanation of the differences between processors.)

Reply:
@kayceesrk The CI failure at `multicore_systhreads.ml` appears to come from here, let me know if you have an idea.

Reply:
Addendum: `caml_bt_is_in_blocking_section` was not meant as a way to check whether we are the backup thread or not.
The purpose is rather to check around the masterlock what is the proper course of action. (ie. is there a thread waiting to run? should we signal the backup thread if not?)

Reply:
Thanks for investigating. Naive question (I did not look at the code), is it not possible to test backup_thread_message for the two values? Otherwise a new thread-local variable is fine too.

Reply:
@gadmm I don't think this would work: because either the backup thread is processing (`IN_BLOCKING_SECTION`), or is about to yield, or already waiting on the domain lock (`ENTERING_OCAML`). (and then there's also `BT_TERMINATING` and `BT_INIT` for lifecycle purposes.)
This would probably always return true for any thread calling it.

I think we do need to expose a way to tell if a thread is a backup thread, though there may be a nicer solution that eludes me.